src/components/TagForm/TagForm.tsx

```diff
@@ -1,41 +1,50 @@
 import { LoaderOverlay } from 'components/LoaderOverlay/LoaderOverlay';
-import { ReactElement } from 'react';
+import { ReactElement, useMemo } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPen } from '@fortawesome/free-solid-svg-icons';
 import { useTagFormContext } from './TagFormContext';
 import { Button } from 'components/Buttons/Button';
-import cn from 'classnames';
+import * as yup from 'yup';
+import { Input } from 'components/Inputs/Input';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { TextArea } from 'components/Inputs/TextArea';
+import { useProducts } from 'hooks/useProducts';
+import { Checkbox } from 'components/Inputs/Checkbox';
+import { Label } from 'components/Inputs/Label';
 
-export const TagForm = (): ReactElement => {
-  const formContext = useTagFormContext();
-  const [formTag, setFormTag] = formContext.useFormTag;
-  const subHeadingClass = 'italic text-[#A2AAB6] font-poly';
+const formSchema = yup.object({
+  name: yup.string().required('Tag name is required'),
+  desc: yup.string().optional(),
+  products: yup.array().of(yup.number()).optional(),
+});
+
+type FormSchema = yup.InferType<typeof formSchema>;
 
-  //Placeholder Data
-  interface IProduct {
-    id: number;
-    name: string;
-  }
-  const allProducts: IProduct[] = [
-    { id: 0, name: 'Burger' },
-    { id: 1, name: 'Chips' },
-    { id: 2, name: 'Salad' },
-  ];
+export const TagForm = (): ReactElement => {
+  const tagForm = useTagFormContext();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormSchema>({
+    resolver: yupResolver(formSchema),
+    defaultValues: {
+      name: tagForm.initialTag?.name,
+      desc: tagForm.initialTag?.description,
+      products: tagForm.initialTag?.products ?? [],
+    },
+  });
+  const { products } = useProducts();
 
-  const toggleProduct = (event: React.ChangeEvent<HTMLInputElement>, toggleProductId: number): void => {
-    if (event.target.checked) {
-      setFormTag({
-        ...formTag,
-        products: formTag.products.concat([toggleProductId]),
-      });
-    } else {
-      setFormTag({
-        ...formTag,
-        products: formTag.products.filter((productId) => productId !== toggleProductId),
-      });
-    }
+  const submitForm = (data: FormSchema): void => {
+    console.log(data);
   };
 
+  const filteredProducts = useMemo(() => {
```

Comment:
do you think we should handle this within the hooks? I've done a similar thing in my previous ticket, but can't help think there's a better way

Reply:
yeah I don't really see a reason why we shouldn't - we don't give the user to choose the filters so we may as well have it all pre-filtered

something like this works...

```
export const useProducts = ({ filterRetired }: Props): UseProductsResult => {
  const products = useQuery<ProductDto[]>(['items'], productService.getAll, {
    select: (data) => (filterRetired ? data.filter((val) => !val.retired) : data),
  });
  ...
}
```

should I implement this throughout for this PR or do you wanna make it a separate ticket?

Reply:
yeah I'll make a separate ticket after this is merged